src/librustc_mir/transform/simplify_cfg.rs

```diff
@@ -65,25 +44,70 @@ impl SimplifyCfg {
             Some(target)
         }
 
+        let mut predecessor_map = util::build_predecessor_map(mir);
+
         let mut changed = false;
-        for bb in mir.all_basic_blocks() {
+        let mut seen = vec![false; mir.basic_blocks.len()];
```

Comment:
I don't think they are large enough for it to make a big difference. However, they might be small enough that using a `u64` or even a `u32` with no allocation might work (with a fallback to `Vec` or `BitVec`).

Regardless, I think such a change should be in a separate PR.